lib/dataset-tools/parsers.ts

```diff
@@ -206,14 +210,19 @@ class DialogueSerializer extends Stream.Transform {
             const turn = dlg.turns[i];
             if (i > 0) {
                 if (this._annotations)
-                    this._pushMany(this._prefixLines(turn.context!, 'C: '));
-                this.push('A: ' + turn.agent + '\n');
+                    this._pushMany(this._prefixLines(turn.context, 'C: '));
+                this._pushMany(this._prefixLines(turn.agent, 'A: '));
                 if (this._annotations)
-                    this._pushMany(this._prefixLines(turn.agent_target!, 'AT: '));
+                    this._pushMany(this._prefixLines(turn.agent_target, 'AT: '));
                 if (this._annotations && turn.intermediate_context)
                     this._pushMany(this._prefixLines(turn.intermediate_context, 'C: '));
+
+                if (turn.rate)
+                    this._pushMany(this._prefixLines(turn.rate, '#! '));
```

Comment:
If we're introducing `#!` to mean "comments that are actually structured data", maybe we should have some sort format inside them, maybe a dictionary or something? So you'd have:
```
#! rate=good timestamp=2021-02-05T... comment=...
```
(it can be even json, although it would be a bit less readable)
(I'm including "comment" in there because I don't quite like conflating the user's comments on the agent reply with any free-form comments of whoever made the file)